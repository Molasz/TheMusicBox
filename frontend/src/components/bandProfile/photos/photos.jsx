import React from 'react';
import './photos.scss';

import PhotoIcon from '@material-ui/icons/PhotoLibrary';

function photos(props) {
  return (
    <section className='photo'>
      <div className='photo__title'>
        <PhotoIcon className='title__icon' />
        <h1 className='title__text'>Photos and videos</h1>
      </div>
      <div className='photo__main'>
        {props.data.map((element, i) => {
          return (
            <img src={element} alt={`nÂº${i}`} key={i} className='main__item' />
          );
        })}
      </div>
    </section>
  );
}

export default photos;
