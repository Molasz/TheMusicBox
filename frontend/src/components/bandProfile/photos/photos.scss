.photo {
  .photo__title {
  }
  .photo__main {
    display: flex;
    flex-wrap: wrap;

    .main__item {
      margin: 20px;
      width: 150px;
      filter: grayscale(100%);
    }
    .main__item:hover {
      filter: grayscale(0);
    }
  }
}
