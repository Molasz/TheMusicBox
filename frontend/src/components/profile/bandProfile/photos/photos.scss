$color-detail: #f55110;

.photo {
  width: 100%;
  .photo__title {
    display: flex;
    align-items: center;

    .title__icon {
      font-size: 30px;
      color: $color-detail;
      margin-right: 10px;
    }
  }
  .photo__main {
    display: flex;
    flex-wrap: wrap;

    .main__item {
      margin: 20px;
      margin-left: 0;
      width: 180px;
      filter: grayscale(100%);
    }
    .main__item:hover {
      filter: grayscale(0);
    }
  }

  .photo__show {
    display: flex;
    flex-direction: column;
    align-items: center;
    .show__main {
      display: flex;
      align-items: center;

      padding: 20px 50px;

      .main__left {
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;

        .left__back {
          justify-self: start;
          font-size: 30px;

          &:hover {
            color: $color-detail;
          }
        }
      }

      .main__photo {
        width: 100%;
        max-width: 600px;

        max-height: 600px;
      }

      .arrow {
        margin: 0 10px;
        font-size: 30px;

        &:hover {
          color: $color-detail;
        }
      }
    }

    .show__footer {
      display: flex;
      flex-wrap: wrap;
      align-content: center;

      .footer__item {
        height: 100px;
        filter: grayscale(100%);
        margin: 5px;

        &:hover {
          filter: grayscale(0);
        }
      }
      .color {
        filter: grayscale(0);
      }
    }
  }
}

@media (min-width: 992px) {
  .photo {
    width: 80%;
  }
}
