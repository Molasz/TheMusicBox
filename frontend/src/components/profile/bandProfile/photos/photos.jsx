import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import './photos.scss';

import {
  showPhoto,
  createPhoto,
  deletePhoto
} from '../../../../redux/actions/bandActions';
import { clearImage, uploadImage } from '../../../../redux/actions/infoActions';

import PhotoIcon from '@material-ui/icons/PhotoLibrary';
import ArrowRight from '@material-ui/icons/ArrowForwardIos';
import ArrowLeft from '@material-ui/icons/ArrowBackIos';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import AddPhotoAlternateIcon from '@material-ui/icons/AddPhotoAlternate';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';

function Photos({ photos, editInfo, photo, image, band, dispatch }) {
  const [newPhotoInput, setNewPhotoInput] = useState();

  useEffect(() => {
    if (image.identifier === 'new-photo') {
      dispatch(createPhoto(band._id, image.path));
      dispatch(clearImage());
    }
  }, [image]);

  return (
    <section className='photo'>
      <div className='photo__title'>
        <PhotoIcon className='title__icon' />
        <h1 className='title__text'>Photos and videos</h1>
      </div>
      {photo == null ? (
        <div className='photo__main'>
          {photos.map((element, i) => {
            return (
              <img
                src={element}
                alt={i}
                key={i}
                className='main__item'
                onClick={() => dispatch(showPhoto(i))}
              />
            );
          })}

          {editInfo.name && (
            <div className='main__new-photo'>
              <AddPhotoAlternateIcon
                className='new-photo__icon'
                onClick={() => {
                  if (editInfo.name) newPhotoInput.click();
                }}
              />

              <input
                type='file'
                name='file'
                style={{ display: 'none' }}
                ref={(fileInput) => setNewPhotoInput(fileInput)}
                onChange={(event) => {
                  const img = new FormData();
                  img.append('file', event.target.files[0]);
                  dispatch(uploadImage(band._id, img, 'new-photo'));
                }}
              />
            </div>
          )}
        </div>
      ) : (
        <div className='photo__show'>
          <div className='show__top'>
            <ArrowBackIcon
              className='top__icon'
              onClick={() => dispatch(showPhoto(null))}
            />

            {editInfo.name && (
              <HighlightOffIcon
                className='top__icon'
                onClick={() => {
                  dispatch(deletePhoto(band._id, band.photos[photo]));
                  dispatch(showPhoto(null));
                }}
              />
            )}
          </div>
          <div className='show__main'>
            <div className='main__left'>
              <ArrowLeft
                id='left-arrow'
                className='arrow'
                onClick={() =>
                  dispatch(
                    showPhoto(photo === 0 ? photos.length - 1 : photo - 1)
                  )
                }
              />
            </div>
            <img src={photos[photo]} alt='zoom' className='main__photo' />
            <ArrowRight
              id='right-arrow'
              className='arrow'
              onClick={() =>
                dispatch(showPhoto(photo === photos.length - 1 ? 0 : photo + 1))
              }
            />
          </div>
          <div className='show__footer'>
            {photos.map((element, i) => {
              return (
                <img
                  src={element}
                  alt={i}
                  key={i}
                  id={`photo$i`}
                  className={`footer__item ${i === photo ? 'color' : ''}`}
                  onClick={() => dispatch(showPhoto(i))}
                />
              );
            })}
          </div>
        </div>
      )}
    </section>
  );
}

function mapStateToProps(state) {
  return {
    editInfo: state.bandReducer.editInfo,
    photo: state.bandReducer.showPhoto,
    band: state.bandReducer.band,
    image: state.infoReducer.image
  };
}

export default connect(mapStateToProps)(Photos);
export { Photos };
