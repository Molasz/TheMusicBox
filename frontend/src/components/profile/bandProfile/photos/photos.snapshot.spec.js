import React from 'react';
import renderer from 'react-test-renderer';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

import { Photos } from './photos';

describe('Photos snapshot', () => {
  const photos = ['testPhoto.com', 'mockPhoto.com'];
  const photo = 1;
  const dispatch = jest.fn();
  it('Should match ', () => {
    const tree = renderer.create(<Photos photos={photos} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should match ', () => {
    const tree = renderer.create(<Photos photos={photos} photo={photo} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should call dispatch when click', () => {
    const document = shallow(
      <Photos photos={photos} photo={photo} dispatch={dispatch} />
    );
    const button = document.find('.left__back');
    button.simulate('click', {});

    expect(dispatch.call).truthy;
  });

  it('Should call dispatch when click', () => {
    const document = shallow(
      <Photos photos={photos} photo={photo} dispatch={dispatch} />
    );
    const button = document.find('#right-arrow');
    button.simulate('click', {});

    expect(dispatch.call).truthy;
  });

  it('Should call dispatch when click', () => {
    const document = shallow(
      <Photos photos={photos} photo={photo} dispatch={dispatch} />
    );
    const button = document.find('#left-arrow');
    button.simulate('click', {});

    expect(dispatch.call).truthy;
  });

  it('Should call dispatch when click', () => {
    const document = shallow(
      <Photos photos={photos} photo={0} dispatch={dispatch} />
    );
    const button = document.find('#right-arrow');
    button.simulate('click', {});

    expect(dispatch.call).truthy;
  });

  it('Should call dispatch when click', () => {
    const document = shallow(
      <Photos photos={photos} photo={0} dispatch={dispatch} />
    );
    const button = document.find('#left-arrow');
    button.simulate('click', {});

    expect(dispatch.call).truthy;
  });
});
