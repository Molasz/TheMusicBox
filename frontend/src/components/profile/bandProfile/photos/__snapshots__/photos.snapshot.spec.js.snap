// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos snapshot Should match  1`] = `
<section
  className="photo"
>
  <div
    className="photo__title"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root title__icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
      />
    </svg>
    <h1
      className="title__text"
    >
      Photos and videos
    </h1>
  </div>
  <div
    className="photo__main"
  >
    <img
      alt={0}
      className="main__item"
      onClick={[Function]}
      src="testPhoto.com"
    />
    <img
      alt={1}
      className="main__item"
      onClick={[Function]}
      src="mockPhoto.com"
    />
  </div>
</section>
`;

exports[`Photos snapshot Should match  2`] = `
<section
  className="photo"
>
  <div
    className="photo__title"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root title__icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4l2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
      />
    </svg>
    <h1
      className="title__text"
    >
      Photos and videos
    </h1>
  </div>
  <div
    className="photo__show"
  >
    <div
      className="show__main"
    >
      <div
        className="main__left"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root left__back"
          focusable="false"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root arrow"
          focusable="false"
          id="left-arrow"
          onClick={[Function]}
          viewBox="0 0 24 24"
        >
          <path
            d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
          />
        </svg>
      </div>
      <img
        alt="zoom"
        className="main__photo"
        src="mockPhoto.com"
      />
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root arrow"
        focusable="false"
        id="right-arrow"
        onClick={[Function]}
        viewBox="0 0 24 24"
      >
        <path
          d="M5.88 4.12L13.76 12l-7.88 7.88L8 22l10-10L8 2z"
        />
      </svg>
    </div>
    <div
      className="show__footer"
    >
      <img
        alt={0}
        className="footer__item "
        id="photo$i"
        onClick={[Function]}
        src="testPhoto.com"
      />
      <img
        alt={1}
        className="footer__item color"
        id="photo$i"
        onClick={[Function]}
        src="mockPhoto.com"
      />
    </div>
  </div>
</section>
`;
